### ISSUE TITLE
Running .NET MAUI 8 app on real iOS device getting error - Undefined symbols for architecture arm64

### ISSUE THREAD

**Issue by khambley** [marcojak/MauiMTAdmob#60]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/17.0.8478/targets/Xamarin.Shared.Sdk.targets(3,3): Error: clang++ exited with code 1:
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_UMPConsentForm", referenced from:
      objc-class-ref in registrar.o
  "_OBJC_CLASS_$_UMPConsentInformation", referenced from:
      objc-class-ref in registrar.o
  "_OBJC_CLASS_$_UMPDebugSettings", referenced from:
      objc-class-ref in registrar.o
  "_OBJC_CLASS_$_UMPRequestParameters", referenced from:
      objc-class-ref in registrar.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

This build error only occurs when I try to test on my physical iOS devices running iOS 17.2. The plugin works fine on the iOS simulators. I'm building and running it on a Mac in VS for Mac, 17.6.8. Android works fine on simulators and physical devices. I'm running Xcode 15.0.1. I see there is an update for Xcode to 15.2. I'll try updating it and see if that solves it.

**Comment by erossini** [comment]
I get the same error and I also opened a post on [Stackoverflow](https://stackoverflow.com/questions/77905778/net8-maui-and-deploy-for-testing-to-a-real-device) about it.

I'm not sure is related to the component.

**Comment by marcojak** [comment]
I've tried on an iPad and it was working fine but let me try again with this version and I'll update you asap.

**Comment by khambley** [comment]
Yeah, I updated my Xcode to 15.2 and still getting the error above. Not sure what I'm missing here.

**Comment by marcojak** [comment]
Just tested again and now I get the same issue.
I'm investigating it...

**Comment by marcojak** [comment]
Not sure if this matters,
however, I've updated Visual Studio on my PC, I've updated VS on the mac (this might be less relevant) and run the sample on my iPad...then it worked without getting the Arm64 error.

Could you try to update VS and test it again?

**Comment by erossini** [comment]
I have updated my VS but I still get the error. As a side not, I got another issue because I couldn't deploy the app to the simulator. For that, I added to the project file those lines

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net8.0-ios|AnyCPU'">
		<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
	</PropertyGroup>

**Comment by khambley** [comment]
> Not sure if this matters, however, I've updated Visual Studio on my PC, I've updated VS on the mac (this might be less relevant) and run the sample on my iPad...then it worked without getting the Arm64 error.
> 
> Could you try to update VS and test it again?

My VS is up to date as well. What Apple chip does your Mac have? I'm running on an M1. I wonder if it needs to compile on an M2? Just a shot in the dark.

**Comment by erossini** [comment]
My iMac is an M1

**Comment by khambley** [comment]
> My iMac is an M1. I have bought it last year only for testing the apps 😉

Mine is too. That question was directed @marcojak I wonder if they have an M2.

**Comment by khambley** [comment]
I just tried building it on my Macbook Pro which has an M2 chip and same error. I've tried testing it on an iPhone 13 Pro Max and an iPhone SE (3rd gen) both are iOS 17.2.

**Comment by khambley** [comment]
@marcojak I've got a PC, I'm going to give it a try on there, maybe that's the issue.

**Comment by marcojak** [comment]
So, currently I have a MacBook Pro with an i7.

I don't think the Mac architecture is related to the issue as I was able to experience it and then again after I rebuilt this sample app, it worked.
While I was doing my test, in the configuration manager I set up ARM64 instead of any.
Not sure if that could be the reason why it worked. What I can do, if you are happy to try, is to send you a new version of the library to see if that works for you.

**Comment by erossini** [comment]
Sure

**Comment by khambley** [comment]
Yeah, we don't have a configuration manager option on Visual Studio for Mac. I believe it defaults to x64 architecture, I'll see if I can add arm64 in the proj file.

**Comment by khambley** [comment]
I'm at a loss. I tried building it on a PC that was paired to my Mac. And same error when deploying to a physical iOS device, iPhone 13 Pro Max. I did try setting the CPU to arm64 directly and I'm getting the same error. I'm sorry to say, I'll have to remove MauiMTAdmob and ads for now from iOS build until this is fixed. Bummer!

**Comment by marcojak** [comment]
[Plugin.MauiMTAdmob.1.1.1.zip](https://github.com/marcojak/MauiMTAdmob/files/14114808/Plugin.MauiMTAdmob.1.1.1.zip)

Here is the updated version. Could you please try it?

There are no changes but hopefully this should support ARM64.

To make it work:

1. Remove the plugin from your app
2. Extract and add this package to your local nugets (if you don't have it already configured, you can do it from VS).
3. Remove the plugin from the cache **(This is required, otherwise, the app will use the package from your cache).**
4. Add the package to the app again

Please, let me know if it works.

**Comment by marcojak** [comment]
> I'm at a loss. I tried building it on a PC that was paired to my Mac. And same error when deploying to a physical iOS device, iPhone 13 Pro Max. I did try setting the CPU to arm64 directly and I'm getting the same error. I'm sorry to say, I'll have to remove MauiMTAdmob and ads for now from iOS build until this is fixed. Bummer!

Could you try the new version I added in the comment just up here?

**Comment by erossini** [comment]
No, unfortunately some error 😒

**Comment by marcojak** [comment]
Which version of xcode do you have?

**Comment by erossini** [comment]
15.2

<img width="642" alt="Screenshot 2024-01-31 at 17 39 44" src="https://github.com/marcojak/MauiMTAdmob/assets/9497415/22731f7d-6f41-49c6-83f4-2cb260486a7d">

**Comment by marcojak** [comment]
I'm using 15.1.

Let me update it to 15.2 and try again.
It might take a while as my machine is pretty old, but I'll update you as soon as I can try it.

**Comment by marcojak** [comment]
While I continue my tests, could you add this package to your iOS app?

<PackageReference Include="MT.Xamarin.Google.iOS.UserMessagingPlatform" Version="1.1.0.2"/>

This contains the missing references....maybe referencing it directly might solve the issue...

**Comment by erossini** [comment]
> While I continue my tests, could you add this package to your iOS app?
> 
> This contains the missing references....maybe referencing it directly might solve the issue...

Sorry, what package? If you refer to the zip file, yes, I added to my project and I get the same issue

**Comment by marcojak** [comment]
this one: => "MT.Xamarin.Google.iOS.UserMessagingPlatform" Version="1.1.0.2"

sorry it didn't appear in the previous comment

**Comment by khambley** [comment]
So, I downloaded and installed MT.Xamarin.Google.iOS.UserMessagingPlatform" Version="1.1.0.2, cleaned, rebuilt and tried to run on iOS physical device, still getting error.
I tried this workaround, in the cproj file: per this link https://stackoverflow.com/questions/76618970/appdynamics-agent-for-maui-symbol-not-found-for-architecture-x86-64

<Project Sdk="Microsoft.NET.Sdk">
  ...
  <PropertyGroup Condition="$(TargetFramework.Contains('-ios'))">
    <MTouchExtraArgs>--registrar:static</MTouchExtraArgs>
  </PropertyGroup>
  
</Project>
And it first it looked promising, I was able to build the iOS app and it started to launch but then it got to the launch screen and crashed. UGH!! I'll keep trying things, about to pull my hair out!

**Comment by marcojak** [comment]
I've moved from Visual studio connected to a mac to VS for mac on my mac with i7.
Now I again see the same error (so at least it's not an issue of i7 vs M1 or M2...
It's painful as my mac takes ages to build so it's quite slow, but I'll try to find a solution asap.

**Comment by khambley** [comment]
> Ok, I think I've found a "temporary" solution.
> 
> In your ios project import this package: "Xamarin.Google.iOS.UserMessagingPlatform" Version="1.1.0.1"
> 
> I'm investigating it further to see why we see this error.

So I did that and it completely hosed my app. It won't load at all. I re-cloned it from my repo and pulled it down prior to adding your plugin. I created a new MAUI project on my PC and added in your plugin and the setup for iOS including the license. I got it to deploy once to my iOS physical device but the ads did not show up. I then tried again and I got all kinds of errors, like 

2022-08-22 23:14:34.618 Xamarin.PreBuilt.iOS[94921:6409975] Could not resolve assembly Microsoft.VisualStudio.DesignTools.TapContract, Version=17.0.0.0, Culture=neutral, PublicKeyToken=null. Details: Could not load file or assembly '/var/mobile/Containers/Data/Application/6B3B6CF6-2A5A-452F-956D-74D8A66941EA/Documents/Lofty.Logbook.content/Microsoft.VisualStudio.DesignTools.TapContract.dll' or one of its dependencies. 

It runs on the PC connected to my Mac on the iOS simulators for 17.2. But not physical iOS devices. It's too buggy and unreliable at this point. I'm going to have to find another solution, maybe this tutorial looks promising... ? [Admob](https://startdebugging.net/2023/11/how-to-add-admob-to-your-maui-app/)

**Comment by erossini** [comment]
> Ok, I think I've found a "temporary" solution.
> 
> In your ios project import this package: "Xamarin.Google.iOS.UserMessagingPlatform" Version="1.1.0.1"
> 
> I'm investigating it further to see why we see this error.

I added this package and now I can deploy on my physical iPhone the application. Thank you
The problem is that the advertisements are not displaying at all, in particular banners and interstitial rewards.

**Comment by marcojak** [comment]
I'm glad it works...
I've noticed something else that doesn't work as it should on iOS physical device and this could be the reason why you don't see ads...I hope to send you a new version to test by the end of the morning so you can test it.

In general, If you are using test IDs, then you should be able to see ads even if not on the store yet.

**Comment by erossini** [comment]
There is another thing I noticed but in both platforms. I should open another ticket. Briefly, with the interstitial rewards I call

```
CrossMauiMTAdmob.Current.ShowRewarded();
```

and then the app is waiting for one of the events to continue. The problem here is that the rewarded is not showing at all and not event is fired.

**Comment by marcojak** [comment]
> There is another thing I noticed but in both platforms. I should open another ticket. Briefly, with the interstitial rewards I call
> 
> ```
> CrossMauiMTAdmob.Current.ShowRewarded();
> ```
> 
> and then the app is waiting for one of the events to continue. The problem here is that the rewarded is not showing at all and not event is fired.

Please, open another ticket and I'll have a look and fix it if needed...just to be sure, before calling show rewarded, have you loaded the ad? did you get an event that the ad was loaded successfully (or failed to load)?

**Comment by marcojak** [comment]
Could you please try this one?
If everything is ok, I'll update on nuget.

[Plugin.MauiMTAdmob.1.1.2.zip](https://github.com/marcojak/MauiMTAdmob/files/14123989/Plugin.MauiMTAdmob.1.1.2.zip)

**Comment by erossini** [comment]
Do I need to remove the package `Xamarin.Google.iOS.UserMessagingPlatform`?

**Comment by marcojak** [comment]
No, for now keep it.
Still trying to fix the issue...

**Comment by erossini** [comment]
I quickly tried this version and it seems working.

**Comment by marcojak** [comment]
That's great.
I'll see if I can do something to avoid that import in the iOS project and then I'll release it to everyone

**Comment by erossini** [comment]
Yes, it is working. Thank you so much!

**Comment by khambley** [comment]
I updated your plugin to 1.1.2. The license and app Id is correctly set in the Info.plist and the AppDelegate. I'm getting an error and the app crashes on iOS physical device:
 An error occurred: 'Object reference not set to an instance of an object.'. Callstack: '   at Plugin.MauiMTAdmob.Platforms.iOS.UMPImplementation..ctor(MauiMTAdmob mauiMTAdmob, Boolean tagForUnderAgeOfConsent, String testDeviceId, DebugGeography geography)
   at Plugin.MauiMTAdmob.MauiMTAdmob.Init(String license, Boolean tagForUnderAgeOfConsent, String testDeviceId, DebugGeography geography)
   at Maui3Demo.AppDelegate.FinishedLaunching(UIApplication application, NSDictionary launchOptions) in C:\Projects\Maui3Demo\Maui3Demo\Platforms\iOS\AppDelegate.cs:line 18
   at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName) in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:line 58
   at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass) in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:line 94
   at Maui3Demo.Program.Main(String[] args) in C:\Projects\Maui3Demo\Maui3Demo\Platforms\iOS\Program.cs:line 13
   at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)'
The app has been terminated.
2024-02-01 08:35:48.128 Xamarin.PreBuilt.iOS[1380:324948] MTAdmob correctly licensed

**Comment by erossini** [comment]
It is working for me. For iOS I set

```csharp
    public override bool FinishedLaunching(UIApplication application, NSDictionary launchOptions)
    {
        CrossMauiMTAdmob.Current.Init(Constants.MauiMTAdmobLicence);
        return base.FinishedLaunching(application, launchOptions);
    }
```

if it could be helpful

**Comment by khambley** [comment]
> It is working for me. For iOS I set
> 
> ```cs
>     public override bool FinishedLaunching(UIApplication application, NSDictionary launchOptions)
>     {
>         CrossMauiMTAdmob.Current.Init(Constants.MauiMTAdmobLicence);
>         return base.FinishedLaunching(application, launchOptions);
>     }
> ```
> 
> if it could be helpful

Tried that, same error

**Comment by marcojak** [comment]
Have you removed from your iOS project Xamarin.Google.iOS.UserMessagingPlatform?

Out of curiosity, can I ask you to remove the app from the simulator and device, rebuild your project and test it again?

Does the sample work for you?

I'll do some further tests and see if I can spot any null references and if needed release a new version

**Comment by erossini** [comment]
@khambley you are right! In the Simulator I get the same error

![image](https://github.com/marcojak/MauiMTAdmob/assets/9497415/9f5151bf-a5ac-49ee-80aa-8b697f0ba9cb)

**Comment by marcojak** [comment]
Just the simulator?
This might give me an idea...sorry about this...

**Comment by erossini** [comment]
OK, now I tried to deploy to the real device and I have other errors because I'm doing same changing but I can see something that I don't like anymore 😁

![image](https://github.com/marcojak/MauiMTAdmob/assets/9497415/a1f1f1eb-339f-4b58-b7ed-fc8eb3d98e3b)

This happens after replacing the zip package with the package from the NuGet online.

**Comment by erossini** [comment]
OK, now I tried to deploy to the real device and I have other errors because I'm doing same changing but I can see something that I don't like anymore 😁

![image](https://github.com/marcojak/MauiMTAdmob/assets/9497415/a1f1f1eb-339f-4b58-b7ed-fc8eb3d98e3b)

This happens after replace the zip package with the package from the NuGet online.

**Comment by erossini** [comment]
The cards are not your fault, I agree 😁 but if you want to fix them...

**Comment by erossini** [comment]
I happily fixed my cards problem. Still the issue with `Xamarin.Google.iOS.UserMessagingPlatform`

![image](https://github.com/marcojak/MauiMTAdmob/assets/9497415/9b3adbbe-afff-4bdd-a23d-0fcdce1d6d3f)

**Comment by khambley** [comment]
I haven't tried your sample. I started with a new clean MAUI .net 8 project on my PC connected to my Mac. I did create a new MAUI project on my Mac as well and it wouldn't let me load your nuget package. It said it was incompatible with maccatalyst. It seems to install okay but it won't recognize the Plugin.MauiMTAdmob namespace so I can't call it in the project. It won't build. All of my software is up to date with the latest stable versions, Xcode 15.2, macOs Sonoma 14.3, Visual Studio for Mac 17.6.8

**Comment by marcojak** [comment]
I've released version 1.1.3 of the plugin to avoid the crash on the simulator.
I'm not sure why the consent sharedinstance is null on a simulator and works fine on the real device but most important I'm not sure why it used to work fine (on the simulator) and now it doesn't.
I've tried to compile a previous commit but nothing. And given that the code used to work and now the same code doesn't work, I'm inclined to think that this happened after I've updated something (I've updated VS on Windows, VS for MAC and XCode).
I'll continue to investigate and see what I can do to make it work on the simulator

**Comment by erossini** [comment]
Thank you so much!

**Comment by marcojak** [comment]
I've just released version 1.2.0
The news is that it supports App Open Ads for Android and iOS.

Let me know if this version works well and you can normally compile your code...

**Comment by khambley** [comment]
So I cannot install the update for the plugin 1.2.0 on VS for Mac. 17.6.8 (build 400). Error Message: 
Package Plugin.MauiMTAdmob 1.2.0 is not compatible with net8.0-ios17.0 (.NETCoreApp,Version=v8.0). Package Plugin.MauiMTAdmob 1.2.0 supports:
  - net8.0-android34.0 (.NETCoreApp,Version=v8.0)
  - net8.0-ios17.2 (.NETCoreApp,Version=v8.0)
  - net8.0-maccatalyst17.2 (.NETCoreApp,Version=v8.0)
  - net8.0-windows10.0.19041 (.NETCoreApp,Version=v8.0)
 I have the latest stable XCode version 15.2. My iOS Simulators are iOS 17.2 and my physical devices are 17.2. My dotnet version is 8.0.100 which from what I can tell is the latest stable version. 
I've tried everything I can think of to upgrade the net8.0-ios17.0 to 17.2, am at a loss. Does anyone know how to update that framework to 17.2?

**Comment by marcojak** [comment]
Let me check if I can change it...I'm adding back .net7 as well for some
projects that might still be stuck with that so I'll see if I can also
change this compatibility with ios17.0

**Comment by marcojak** [comment]
Do you have an empty sample project that gives you the error and that I can try? If so, send it and I'll see what I can do.

The plugin just targets net8.0-ios

with <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">15.4</SupportedOSPlatformVersion>

So I'm not sure yet why you are seeing the error but I'm investigating it.

**Comment by marcojak** [comment]
Can you check that in your targetFrameworks you have
	<TargetFrameworks>net8.0-android;net8.0-ios;net8.0-maccatalyst</TargetFrameworks>

and not this:
	<TargetFrameworks>net8.0-android;net8.0-ios17.0;net8.0-maccatalyst</TargetFrameworks>

?

**Comment by khambley** [comment]
So I updated my .NET 8 SDK, created a new blank .NET MAUI project on VS for Mac. I installed the latest version of plugin and I get that "Error: clang++ exited with code 1:
Undefined symbols for architecture arm64:
"OBJC_CLASS$_UMPConsentForm", referenced from:
objc-class-ref in registrar.o"... error.
I tried to install the Xamarin.Google.iOS.UserMessagingPlatform version 1.1.0.1. I'm not sure how to target just the iOS project in MAUI so it tried to install in all of them presumably. I get a weird the Name of the assembly does not match the target name in Android. The app installs on iOS device but crashes right after the launch screen.
I installed the Xamarin.Google.iOS.UserMessagingPlatform version 1.2.0.5 and same thing, iOS app installs but crashes. When I try it on Android, it says "failed processing manifest" and fails.

**Comment by marcojak** [comment]
<ItemGroup Condition="'$(TargetFramework)' == 'net8.0-ios'">
	<PackageReference Include="Xamarin.Google.iOS.UserMessagingPlatform" Version="1.1.0.1" />
</ItemGroup>

Don't install Xamarin.Google.iOS.UserMessagingPlatform version 1.2.0.5 , it's totally another package (not from xamarin)

Have a look at my previous comment, maybe it helps.

I've just tested an empty project on VS for Mac, same version as you have, with same xcode and simulator version and it works...make sure your targetframework is correct, once I got the same issue and it was just that...

**Comment by khambley** [comment]
..make sure your target framework is correct... What framework should I target? This is what is in my proj file:

**Comment by khambley** [comment]
This is the error I get when trying to install the Xamarin.Google.ios.UserMessagingPlatform 1.1.0.1
`<PackageReference Include="Xamarin.Google.iOS.UserMessagingPlatform" Version="1.1.0.1" Condition="'$(TargetFramework)' == 'net8.0-ios'"/>`
Error while trying to load the project `'/Users/katherinehambley/Projects/RockBandNameGenerator2/RockBandNameGenerator2.csproj': The '_GoogleUserMessagingPlatformAssemblyName' start tag on line 3 position 6 does not match the end tag of '_GoogleMobileAdsAssemblyName'. Line 3, position 132. /Users/katherinehambley/.nuget/packages/xamarin.google.ios.usermessagingplatform/1.1.0.1/buildTransitive/Xamarin.Google.iOS.UserMessagingPlatform.targets` and I can't load the project at all in VS. I have

**Comment by marcojak** [comment]
It's a known issue with the xamarin plugin...I'm trying to solve this as annoying but meanwhile you can solve following this:

Open the file "xamarin.google.ios.usermessagingplatform\1.1.0.1\buildTransitive\Xamarin.Google.iOS.UserMessagingPlatform.targets"

Replace the line
<_GoogleUserMessagingPlatformAssemblyName>Google.UserMessagingPlatform, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null</_GoogleMobileAdsAssemblyName>

with

<_GoogleUserMessagingPlatformAssemblyName>Google.UserMessagingPlatform, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null</_GoogleUserMessagingPlatformAssemblyName>

That will fix the issue!

**Comment by khambley** [comment]
So, I got it to run and ads show up on the iOS Simulator, however, it launches on my iPhone but crashes. Here is what is in the logs:
`The entry point method could not be loaded due to Could not resolve type with token 01000028 from typeref (expected class 'System.Object' in assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a') assembly:System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a type:System.Object member:(null)`
I've restored all the .NET workloads using dotnet package restore on the CL. Any ideas?

**Comment by khambley** [comment]
OMG, I got it to run on my iPhone, yess!! So, I created a new .NET MAUI project in VS for Mac, installed the plugin, wired it up with a test ad unit id and viola! it works! It's a pretty simple app so I can just migrate my code to this new project. It's was just an app I created for learning .NET MAUI and Admob. Not sure if it's ready for a business app, though. Gonna do some more testing. Thank you for sticking with me!

**Comment by erossini** [comment]
I'm going to update my app today. Let you know 😉

**Comment by marcojak** [comment]
> OMG, I got it to run on my iPhone, yess!! So, I created a new .NET MAUI project in VS for Mac, installed the plugin, wired it up with a test ad unit id and viola! it works! It's a pretty simple app so I can just migrate my code to this new project. It's was just an app I created for learning .NET MAUI and Admob. Not sure if it's ready for a business app, though. Gonna do some more testing. Thank you for sticking with me!

I'm glad everything works. Always happy to help!

**Comment by erossini** [comment]
Sorry, I'm late to the party. I have just installed version 1.2.1 and when I deploy it on a real device, I get this error

> error MT1045: Failed to execute 'devicectl': 'devicectl -j /var/folders/dm/bwmxpbzn6bvdsyy73c_b453w0000gn/T/tmpZlCLVY.tmp device install app --device "Enrico???s iPhone" /Users/enrico/Library/Caches/Xamarin/mtbs/builds/LanguageInUse/1fa03704bb15e35c6f47a701d9d92131e3e0740198296a93338bb3c829bc9cf7/bin/Debug/net8.0-ios/ios-arm64/device-builds/iphone15.2-17.3.1/LanguageInUse.app' returned the exit 
>
> clang++ exited with code 1:
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_UMPConsentForm", referenced from:
      objc-class-ref in registrar.o
  "_OBJC_CLASS_$_UMPConsentInformation", referenced from:
      objc-class-ref in registrar.o
  "_OBJC_CLASS_$_UMPDebugSettings", referenced from:
      objc-class-ref in registrar.o
  "_OBJC_CLASS_$_UMPRequestParameters", referenced from:
      objc-class-ref in registrar.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)	LanguageInUse	C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8004\targets\Xamarin.Shared.Sdk.targets	1559		

Before that, I removed the reference to the `Xamarin.Google.iOS.UserMessagingPlatform`.

**Comment by marcojak** [comment]
For this issue for now you need to reference that package in your iOS project.
I'm still investigating it and see how to solve this.

**Comment by erossini** [comment]
So, I added the package back, but I still have this error.

> error MT1045: Failed to execute 'devicectl': 'devicectl -j /var/folders/dm/bwmxpbzn6bvdsyy73c_b453w0000gn/T/tmp7IdFVh.tmp device install app --device "Enrico???s iPhone" /Users/enrico/Library/Caches/Xamarin/mtbs/builds/LanguageInUse/1fa03704bb15e35c6f47a701d9d92131e3e0740198296a93338bb3c829bc9cf7/bin/Debug/net8.0-ios/ios-arm64/device-builds/iphone15.2-17.3.1/LanguageInUse.app' returned the exit code 1.

It works in the Simulator. So, I tried to downgrade to version 1.2.0 with the same error. Maybe it is not related to your component...

**Comment by marcojak** [comment]
Mhh...i don't recognise this error.
Currently I don't think it's related to this plugin.
Anyway I would try to clean the project and remove bin and obj folders for a clean test.
